types/bluebird/index.d.ts

```diff
@@ -578,42 +581,59 @@ declare class Bluebird<R> implements PromiseLike<R>, Bluebird.Inspection<R> {
   /**
    * Same as calling `Promise.any(thisPromise)`. With the exception that if this promise is bound to a value, the returned promise is bound to that value too.
    */
-  any(): Bluebird<IterableItem<R>>;
+  any<Q>(this: Bluebird<R & Iterable<Q>>): Bluebird<Q>;
+
+  /**
+   * Same as calling `Promise.any(thisPromise)`. With the exception that if this promise is bound to a value, the returned promise is bound to that value too.
+   */
+  any(): Bluebird<never>;
+
+  /**
+   * Same as calling `Promise.some(thisPromise)`. With the exception that if this promise is bound to a value, the returned promise is bound to that value too.
+   * Same as calling `Promise.some(thisPromise)`. With the exception that if this promise is bound to a value, the returned promise is bound to that value too.
+   */
+  some(this: Bluebird<Iterable<{}>>, count: number): Bluebird<R>;
```

Comment:
shouldn't this be `some<Q>(this: Bluebird<R & Iterable<Q>>, count: number): Bluebird<Q>;` ? Or do I misunderstand?

Reply:
Not according to the tests:

```ts
let fooArrProm: Promise<Foo[]>;
fooArrProm = fooArrProm.some(num);
```

It doesn't make sense to me either.

Reply:
Update: I do misunderstand, and the `count` parameter is important.